xapian-applications/omega/query.cc

```diff
@@ -651,24 +652,28 @@ html_strip(const string &str)
     return res;
 }
 
-// FIXME split list into hash or map and use that rather than linear lookup?
-static int word_in_list(const string& word, const string& list)
-{
-    string::size_type split = 0, split2;
+static unordered_map<string, int> word_to_occurrence;
+
+static void build_word_map(const string& list) {
+    string word;
     int count = 0;
```

Comment:
I think this variable should definitely now be called something like `word_index`. Otherwise it lends the impression that we're putting instance counts as values in the map, rather than the (word) index within the list.